api/src/transactions.rs

```diff
@@ -287,18 +288,37 @@ impl Transactions {
         Response::new(self.ledger_info, &txn)
     }
 
-    pub fn signing_message(self, txn: UserTransactionRequest) -> Result<impl Reply, Error> {
+    pub fn signing_message(
+        self,
+        UserCreateSigningMessageRequest {
+            transaction,
+            secondary_signers,
+        }: UserCreateSigningMessageRequest,
+    ) -> Result<impl Reply, Error> {
         let resolver = self.context.move_resolver()?;
         let raw_txn: RawTransaction = resolver
             .as_converter()
-            .try_into_raw_transaction(txn, self.context.chain_id())
+            .try_into_raw_transaction(transaction, self.context.chain_id())
             .map_err(|e| {
                 Error::invalid_request_body(format!("invalid UserTransactionRequest: {:?}", e))
             })?;
 
+        let raw_message = match secondary_signers {
```

Comment:
what's the behavior if secondary_signers is set empty from the REST client? It kinda feels like that condition should likely push this into a traditional RawTransaction or maybe be an error?